src/components/DirectoryTree.tsx

```diff
@@ -281,24 +281,27 @@ export class DirectoryTree extends React.Component<DirectoryTreeProps, {
        * dropped into target or some parent of the target.
        */
       onDragOver(tree: ITree, data: IDragAndDropData, targetElement: File, originalEvent: DragMouseEvent): IDragOverReaction {
-        const items = Array.from(originalEvent.browserEvent.dataTransfer.items);
-        function mimeTypeUploadIsAllowed(type: string) {
-          console.log(type);
-          if (type === "") { // Firefox doesn't show the "application/wasm" mime type.
-            return true;
+        // File being dragged into the browser
+        if (!(data as any).elements) {
+          const items = Array.from(originalEvent.browserEvent.dataTransfer.items);
+          function mimeTypeUploadIsAllowed(type: string) {
+            if (type === "") { // Firefox doesn't show the "application/wasm" mime type.
+              return true;
+            }
+            return fileTypeForMimeType(type) !== FileType.Unknown;
+          }
+          // In Firefox, tree elements get data transfer items with the "text/uri-list" type. This is a
+          // workaround to ignore that behavior.
+          const firefoxWorkaround = !items.find(item => item.type === "text/uri-list");
```

Comment:
nit: `firefoxWorkaround` a bad name -- I suggest `hasItemsUriListType`. Also usage of `items.some` or `items.every` instead of `items.find` will make code more readable.